contrib/win32/win32compat/misc.c

```diff
@@ -489,6 +489,10 @@ strmode(mode_t mode, char *p)
 	case _S_IFIFO:			/* fifo */
 		*p++ = 'p';
 		break;
+	case _S_IFLNK | S_IFDIR:	/* link */
```

Comment:
S_IFDIR is already handled (first case statement at line 475).
So you can remove line 492.

Reply:
Switch statements are not bitwise operations.  Both _S_IFLNK and S_IFDIR are set in the switched variable which needs to be handled as a separate case.

Reply:
@bagajjal Does my comment make sense?  The code was already handling case 0x4000 (S_IFDIR).  I added cases for 0xA000 (_S_IFLNK) and 0xE000 (_S_IFLNK | S_IFDIR).  It appears based on tests with Linux that if something is a link, the mode string should indicate it is a symbolic link regardless as to whether it points to a file or a directory.